src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,230 @@
+  const { isPicture, isName, isIcon, disabled } = props.customProps
+
+  if (isPicture || isName) {
+    return "transparent"
+  } else if (isIcon) {
+    if (state === "default" || disabled) {
+      return getThemeColor(props, "Neutral10")
+    } else if (state === "hover" || state === "focus") {
+      return getThemeColor(props, "Neutral20")
+    }
+  }
+}
+
+const getOpacity = props => {
+  const { isIcon } = props.customProps
+
+  return isIcon ? 1 : 0.4
+}
+
+const AvatarBase = styled.button<IStyled<AvatarBaseProps>>`
+  box-sizing: border-box;
+  width: ${({ customProps }) => customProps.size};
+  height: ${({ customProps }) => customProps.size};
+
+  overflow: hidden;
+  border-radius: 50%;
+  outline: none;
+
+  /* Style for Name Avatar */
+  color: ${props => getThemeColor(props, "Neutral90")};
+  font-size: ${props => getFontSize(props)}rem;
+  line-height: ${props => getLineHeight(props)}rem;
+
+  /* Below Basic Property & corresponding function calls which will play role in :
+   * all three types of Avatar -
+   *  - isPicture
+   *  - isName
+   *  - isIcon
+   * along with thier 3 state & disabled attribute -
+   *  - default
+   *  - hover
+   *  - focus/active
+   */
+  background: ${props => getBackgroundColor(props, "default")};
+  border: ${props => getBorderStyle(props, "default")};
+  img {
+    width: ${({ customProps }) => customProps.size};
+    height: ${({ customProps }) => customProps.size};
+    margin: ${props => getMarginStyle(props)};
+    user-drag: none;
+    user-select: none;
+    transition: transform 120ms;
+  }
+
+  svg {
+    margin: ${props => getMarginStyle(props)};
+    path {
+      fill: ${props => getFillColor(props, "default")};
+    }
+  }
+
+  :hover {
+    img {
+      transform: scale(1.05);
+    }
+    background: ${props => getBackgroundColor(props, "hover")};
+    border: ${props => getBorderStyle(props, "hover")};
+    svg path {
+      fill: ${props => getFillColor(props, "hover")};
+    }
+  }
+  :focus,
+  :active {
+    background: ${props => getBackgroundColor(props, "focus")};
+    border: ${props => getBorderStyle(props, "focus")};
+    svg path {
+      fill: ${props => getFillColor(props, "focus")};
+    }
+    box-shadow: ${props => getBoxShadowStyle(props)};
+  }
+
+  :disabled {
+    opacity: ${props => getOpacity(props)};
+    cursor: not-allowed;
+  }
+`
+
+const Avatar: FC<AvatarProps> = props => {
+  const { picture, size = "24px", name, disabled = false, ...rest } = props
+
+  const isPicture = !!picture
+  const isName = !!name
+  const isIcon = !(isPicture || isName)
+  const avatarTitle = isName ? name : isPicture ? "" : "Unknown User"
+  const customProps = {
+    isPicture,
+    isName,
+    isIcon,
+    size,
+    disabled,
+  }
+
+  return (
+    <AvatarBase
+      data-testid="avatar"
+      customProps={customProps}
+      disabled={disabled}
+      title={avatarTitle}
+      {...rest}>
+      {(isPicture && <img src={picture} />) ||
```

Comment:
How about exporting child components, `Avatar.Picture`, `Avatar.Name` and `Avatar.Icon` ?
This would make it wasier to customize them individually with styled components and we can try to enfore that there can be only one child through typescript,,